Symfony/CS/Fixer/Symfony/PhpdocTrimFixer.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Symfony;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\DocBlock\DocBlock;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Graham Campbell <graham@mineuk.com>
+ */
+class PhpdocTrimFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens->findGivenKind(T_DOC_COMMENT) as $token) {
+            $content = $token->getContent();
+            $content = $this->fixStart($content);
+            $content = $this->fixEnd($content);
+            $token->setContent($content);
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * Make sure the first useful line starts immediately after the first line.
+     *
+     * @param string $content
+     *
+     * @return string
+     */
+    private function fixStart($content)
+    {
+        $doc = new DocBlock($content);
+        $lines = $doc->getLines();
+        $total = count($lines);
+
+        foreach ($lines as $index => $line) {
+            if (!$line->isTheStart()) {
+                if ($line->containsUsefulContent() || $total - $index < 3) {
+                    break;
+                }
+
+                $line->remove();
+            }
+        }
+
+        return $doc->getContent();
+    }
+
+    /**
+     * Make sure the last useful is immediately before after the final line.
+     *
+     * @param string $content
+     *
+     * @return string
+     */
+    private function fixEnd($content)
+    {
+        $doc = new DocBlock($content);
+        $lines = array_reverse($doc->getLines());
+        $total = count($lines);
+
+        foreach ($lines as $index => $line) {
+            if (!$line->isTheEnd()) {
+                if ($line->containsUsefulContent() || $total - $index < 3) {
```

Comment:
ping @GrahamCampbell 
in `fixStart` too. So this should be constant, not magic variable.